pkg/api/pod/warnings.go

```diff
@@ -92,52 +85,23 @@ func warningsForPodSpecAndMeta(fieldPath *field.Path, podSpec *api.PodSpec, meta
 
 	// use of deprecated node labels in selectors/affinity/topology
 	for k := range podSpec.NodeSelector {
-		if msg, deprecated := deprecatedNodeLabels[k]; deprecated {
+		if msg, deprecated := nodeapi.GetLabelDeprecatedMessage(k); deprecated {
 			warnings = append(warnings, fmt.Sprintf("%s: %s", fieldPath.Child("spec", "nodeSelector").Key(k), msg))
 		}
 	}
 	if podSpec.Affinity != nil && podSpec.Affinity.NodeAffinity != nil {
 		n := podSpec.Affinity.NodeAffinity
 		if n.RequiredDuringSchedulingIgnoredDuringExecution != nil {
-			for i, t := range n.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms {
-				for j, e := range t.MatchExpressions {
-					if msg, deprecated := deprecatedNodeLabels[e.Key]; deprecated {
-						warnings = append(
-							warnings,
-							fmt.Sprintf(
-								"%s: %s is %s",
-								fieldPath.Child("spec", "affinity", "nodeAffinity", "requiredDuringSchedulingIgnoredDuringExecution", "nodeSelectorTerms").Index(i).
-									Child("matchExpressions").Index(j).
-									Child("key"),
-								e.Key,
-								msg,
-							),
-						)
-					}
-				}
+			for i, term := range n.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms {
+				warnings = append(warnings, nodeapi.WarningsForNodeSelectorTerm(term, fieldPath.Child("spec", "affinity", "nodeAffinity", "requiredDuringSchedulingIgnoredDuringExecution", "nodeSelectorTerms").Index(i))...)
```

Comment:
construct `fieldPath.Child("spec", "affinity", "nodeAffinity", "requiredDuringSchedulingIgnoredDuringExecution", "nodeSelectorTerms")` once outside the loop